script.js

```diff
@@ -223,22 +223,49 @@ document.querySelector('.modal-close-btn').addEventListener('click', () => {
   document.querySelector('.modal-container').style.display = 'none';
 });
 
-// CONTACT FORM VALIDATION
-
-const emailError = document.querySelector('#errorMessage');
-const emailSent = document.querySelector('#successMsg');
+// CONTACT FORM VALIDATION AND LOCAL STORAGE STARTS HERE //
 
 const form = document.querySelector('#form');
+const emailInput = document.querySelector('#email');
+const fullNameInput = document.querySelector('#name');
+const messageInput = document.querySelector('#message');
+
+window.addEventListener('load', () => {
+  const fullName = localStorage.getItem('fullName');
+  const email = localStorage.getItem('email');
+  const message = localStorage.getItem('message');
 
-function contactValidator(event) {
-  const email = document.querySelector('#email').value;
-  const emailValidation = /[A-Z]/.test(email);
+  if (fullName && email && message) {
+    fullNameInput.value = fullName;
+    emailInput.value = email;
+    messageInput.value = message;
+  }
+});
 
-  if (emailValidation) {
-    emailError.textContent = 'Email Not Sent⚠️, all the letters must be in lower case';
+form.addEventListener('submit', (event) => {
+  const email = emailInput.value.trim();
+  if (email !== email.toLowerCase()) {
     event.preventDefault();
+    const error = document.querySelector('span');
+    error.textContent = 'Please enter your email in lower case.';
+    error.style.color = '#f55800';
+    error.style.fontFamily = 'Inter';
+    error.style.width = '400px';
+    form.appendChild(error);
+    setTimeout(() => {
+      error.remove();
+    }, 1000);
   } else {
-    emailSent.textContent = 'Congratulations, Email sent 👏';
+    const fullName = fullNameInput.value.trim();
+    const message = messageInput.value.trim();
+    const formData = {
+      fullName,
+      email,
+      message,
+    };
+
+    localStorage.setItem('formData', JSON.stringify(formData));
```

Comment:
As the previous reviewer mentioned [here](https://github.com/Peter-Dumbari/Portofolio-Microverse/pull/15#discussion_r1174443621), your form data is preserved when you change it, not when you submit it. So it should like on this GIF:
![f](https://user-images.githubusercontent.com/25887253/233802984-d5e56820-5b73-45d5-a6db-6c060401afa4.gif)

So you need an event listener to listen the changes on input area. Similar to this: `nameInput.addEventListener('input', populateStorage);` Kindly fix and make sure it works like on the GIF.